arcgis-ios-sdk-samples/Features/Display feature layers/DisplayFeatureLayersViewController.swift

```diff
@@ -0,0 +1,189 @@
+        let shapefileAction = UIAlertAction(title: "Shapefile", style: .default) { (_) in
+            self.loadShapefile()
+        }
+        alertController.addAction(shapefileAction)
+        
+        // Add "cancel" item.
+        let cancelAction = UIAlertAction(title: "Cancel", style: .cancel, handler: nil)
+        alertController.addAction(cancelAction)
+        
+        alertController.popoverPresentationController?.barButtonItem = changeFeatureLayerBarButtonItem
+        present(alertController, animated: true)
+    }
+    
+    // MARK: Helper functions
+    
+    /// Load a feature layer with a URL.
+    func loadFeatureServiceURL() {
+        // Initialize the service feature table using a URL.
+        let featureTable = AGSServiceFeatureTable(url: URL(string: "https://sampleserver7.arcgisonline.com/server/rest/services/DamageAssessment/FeatureServer/0")!)
+        // Create a feature layer with the feature table.
+        let featureLayer = AGSFeatureLayer(featureTable: featureTable)
+        // Set the viewpoint to the Los Angeles National Forest.
+        let viewpoint = AGSViewpoint(latitude: 41.773519, longitude: -88.143104, scale: 4513.988705)
+        setMap(featureLayer: featureLayer, viewpoint: viewpoint)
+    }
+    
+    /// Load a feature layer with a portal item.
+    func loadPortalItem() {
+        // Set the portal.
+        let portal = AGSPortal.arcGISOnline(withLoginRequired: false)
+        // Create the portal item with the item ID for the Portland tree service data.
+        let item = AGSPortalItem(portal: portal, itemID: "1759fd3e8a324358a0c58d9a687a8578")
+        // Create the feature layer with the item and layer ID.
+        let featureLayer = AGSFeatureLayer(item: item, layerID: 0)
+        // Set the viewpoint to Portland, Oregon.
+        let viewpoint = AGSViewpoint(latitude: 45.5266, longitude: -122.6219, scale: 6000)
+        setMap(featureLayer: featureLayer, viewpoint: viewpoint)
+    }
+    
+    /// Load a feature layer with a local geodatabase.
+    func loadGeodatabase() {
+        // Instantiate the geodatabase with the file name.
+        geodatabase = AGSGeodatabase(name: "LA_Trails")
+        
+        // Load the geodatabase for feature tables.
+        geodatabase.load { [weak self] (error: Error?) in
+            guard let self = self else { return }
+            if let error = error {
+                self.presentAlert(error: error)
+            } else {
+                // Get the feature table with the name.
+                let featureTable = self.geodatabase.geodatabaseFeatureTable(withName: "Trailheads")!
+                // Create a feature layer with the feature table.
+                let featureLayer = AGSFeatureLayer(featureTable: featureTable)
+                // Set the viewpoint to Malibu, California.
+                let viewpoint = AGSViewpoint(center: AGSPoint(x: -13214155, y: 4040194, spatialReference: .webMercator()), scale: 35e4)
```

Comment:
It can also be the shorthand `AGSPointMakeWebMercator(-13214155, 4040194)`

Reply:
Also I'm seeing different scale representations: decimal, integer, scientific notation. Are they like that by purpose?

Reply:
I would recommend not using `AGSPointMakeWebMercator`. While it does make the code shorter, it doesn't follow the Swift API guidelines. For example, the parameters are unlabeled. Similar convenience functions provided by Apple (like `CGPointMake` for example) aren't even useable from Swift (`'CGPointMake' is unavailable in Swift`).